packages/diagrams/frontend/sirius-components-diagrams/src/renderer/edge/MultiLabelEditableEdge.tsx

```diff
@@ -235,6 +235,17 @@ export const MultiLabelEditableEdge = memo(
       targetY,
     ]);
 
+    useEffect(() => {
+      setEdges((prevEdges) =>
+        prevEdges.map((prevEdge) => {
+          if (prevEdge.id === id && prevEdge.data) {
+            prevEdge.data.edgePath = edgePath;
+          }
+          return prevEdge;
+        })
```

Comment:
You should always consider the prevState as immutable in React, you should create a new node. From the React documentation:

> state is a reference to the component state at the time the change is being applied. It should not be directly mutated. Instead, changes should be represented by building a new object based on the input from state and props.